core/src/main/java/org/elasticsearch/search/rescore/QueryRescorerBuilder.java

```diff
@@ -78,6 +75,25 @@ public QueryRescorerBuilder(QueryBuilder<?> builder) {
     }
 
     /**
+     * Read from a stream.
+     */
+    public QueryRescorerBuilder(StreamInput in) throws IOException {
+        super(in);
```

Comment:
If you leave out the `super(in)` call then it calls `super()` which doesn't read the window. If we didn't have a default constructor (`super()`) then we'd be forced to pick some constructor here.